doc/snmp/snmp-configdb-migration-hld.md

```diff
@@ -0,0 +1,216 @@
+Contact    Contact Email
+---------  -----------------
+Joe        joe@microsoft.com
+admin@str-s6000-acs-11:~$ 
+```
+
+show run snmp location
+```
+admin@str-s6000-acs-11:~$ show run snmp location 
+Location
+----------
+Redmond
+admin@str-s6000-acs-11:~$ 
+```
+
+show run snmp users
+```
+admin@str-s6000-acs-11:~$ show run snmp users
+User    Type    Auth Type    Auth Password    Encryption Type    Encryption Password
+------  ------  -----------  ---------------  -----------------  ---------------------
+Travis  Priv    SHA          TravisAuthPass   AES                TravisEncryptPass
+admin@str-s6000-acs-11:~$
+```
+
+
+# Config Commands
+```
+admin@str-s6000-acs-11:~$ sudo config snmp -h
+Usage: config snmp [OPTIONS] COMMAND [ARGS]...
+
+  SNMP configuration tasks
+
+Options:
+  -?, -h, --help  Show this message and exit.
+
+Commands:
+  community
+  contact
+  location
+  user
+admin@str-s6000-acs-11:~$ 
+```
+
+sudo config snmp community 
+```
+admin@str-s6000-acs-11:~$ sudo config snmp community -h
+Usage: config snmp community [OPTIONS] COMMAND [ARGS]...
+
+Options:
+  -?, -h, --help  Show this message and exit.
+
+Commands:
+  add  Add snmp community
+  del  Add snmp community
+admin@str-s6000-acs-11:~$ 
+```
+
+sudo config snmp contact  
+```
+admin@str-s6000-acs-11:~$ sudo config snmp contact -h
+Usage: config snmp contact [OPTIONS] COMMAND [ARGS]...
+
+Options:
+  -?, -h, --help  Show this message and exit.
+
+Commands:
+  modify  Modify snmp contact
+admin@str-s6000-acs-11:~$ 
+```
+
+sudo config snmp location 
+```
+admin@str-s6000-acs-11:~$ sudo config snmp location -h
+Usage: config snmp location [OPTIONS] COMMAND [ARGS]...
+
+Options:
+  -?, -h, --help  Show this message and exit.
+
+Commands:
+  modify  Modify snmp location
+admin@str-s6000-acs-11:~$ 
+```
+
+sudo config snmp user 
+```
+admin@str-s6000-acs-11:~$ sudo config snmp user -h
+Usage: config snmp user [OPTIONS] COMMAND [ARGS]...
+
+Options:
+  -?, -h, --help  Show this message and exit.
+
+Commands:
+  add  Add snmp user
+  del  Add snmp user
+admin@str-s6000-acs-11:~$ 
+```
+
+# Migration Plan
+In order to move from using the snmp.yml file to using the Redis ConfigDB here are the things the need to be done to move this in a way that is backward compatible. 
+1. Create a python conversion script to take input from snmp.yml file and convert it to Redis ConfigDB format from above schema.
+2. Update Dockerfile.j2 in docker-snmp container to add a line to copy over new python conversion script to "/usr/bin/"
+3. Update the snmpd.conf.j2 jinja template to pull SNMP information from only the Redis ConfigDB
+4. Update start.sh in docker-snmp container to add a line above sonic-cfggen to run python conversion script (which will run everytime this docker container starts) and then comment out the "-y /etc/sonic/snmp.yml" file as this is not needed since we grabbed the snmp.yml info in the python conversion script and have the information available in the Redis ConfigDB. 
```

Comment:
Suggest rewording as follows:

```
4. Update start.sh in docker-snmp container as follows:
    - Add a line above the `sonic-cfggen` call to run the python conversion script from step 1 (which will run every time this docker container starts)
    - Remove the `-y /etc/sonic/snmp.yml` argument from the existing call to sonic-cfggen as it will no longer needed because the python conversion script above will have already loaded that data into Config DB
```

Reply:
Updated.  I like your wording here a lot better.